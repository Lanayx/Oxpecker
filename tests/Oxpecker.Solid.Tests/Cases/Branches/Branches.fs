module Oxpecker.Solid.Tests.Cases.Branches

open Oxpecker.Solid

[<SolidComponent>]
let Test (x: int) : HtmlElement =
    let s, setS = createSignal x

    h1() {
        match s() with
        | 0 -> div().bool("visible", true)
        | _ ->
            match x with
            | 1 ->
                h2() {
                    "Hello"
                }
            | 2 ->
                h3()
            | _ ->
                if x > 10 then
                    h1(id="id1")
                else
                    h1(id="id2").attr("abc", "def") {
                        "Hello"
                    }
    }
