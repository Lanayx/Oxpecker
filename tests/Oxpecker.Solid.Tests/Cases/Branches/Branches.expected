import { createSignal } from "solid-js";

export function Test(x) {
    const patternInput = createSignal(x);
    return <h1>
        {(patternInput[0]() === 0) ? <div bool:visible={true}></div> : ((x === 1) ? <h2>
            Hello
        </h2> : ((x === 2) ? <h3></h3> : ((x > 10) ? <h1 id="id1"></h1> : <h1 id="id2"
            abc="def">
            Hello
        </h1>)))}
    </h1>;
}

